CONTRIBUTING.md

```diff
@@ -29,34 +29,142 @@ This project follows
 
 ## Additional Notes
 
-The following sections give additional tips and best practices when submitting changes.
+The following sections give additional tips and best practices when submitting
+changes.
 
 ### Adding an NPM dependency
 
-When adding a new dependency (especially in `dependency`), you
-need to perform due diligence.
+When adding a new dependency (especially in `dependency`), you need to perform
+due diligence.
 
-1. Look at the transitive dependencies of the package:
-   <http://npm.broofa.com/>
+1. Look at the transitive dependencies of the package: <http://npm.broofa.com/>
 1. Does the package depend on a lot of dependencies?
 1. What's size of the package? Will it cause binary bloat?
-1. Is the package and its transitive dependencies high quality, trust-worthy, and well-maintained projects?
+1. Is the package and its transitive dependencies high quality, trust-worthy,
+   and well-maintained projects?
 1. What version of the package should we depend on?
-1. Are the licenses for the package and its transitive dependencies green? See next section for checking licenses.
+1. Are the licenses for the package and its transitive dependencies green? See
+   next section for checking licenses.
 
 ### Checking NPM dependencies licenses
 
-When adding any new dependencies or updating existing dependencies
-we need to check licenses:
+When adding any new dependencies or updating existing dependencies we need to
+check licenses:
 
 ```console
 npm run lint-license
 ```
 
+### Local development
+
+When developing locally, you often want to use the the local modified version of
+NPM packages (`kpt-functions` and `create-kpt-functions`) used by function
+packages (e.g. `demo-functions`). You can do the following:
+
+```console
+cd ts/demo-functions
+# May need to run as `sudo` depending on how you intalled NPM
+npm link ../kpt-functions
+npm run build
+```
+
 ### Releases
 
-Release workflows are triggered by creating Releases in the GitHub UI with a tag following the
-format:
+This repo contains two NPM packages that are published to the NPM registry.
+
+#### `kpt-functions`
+
+This is the library used by all TS SDK functions.
+
+##### NPM registry
+
+https://www.npmjs.com/package/kpt-functions
+
+##### Release candidates
+
+Before releasing major changes, you first want to create a `rc` release.
```

Comment:
Whenever the release person doesn't feel 100% confident, he|she should do a `rc` release. I'd suggest add something like "When in doubt, do a `rc` release."